### PR TITLE
Problem: missing api of get_backup_mnemonics (fix #117)

### LINKED ISSUE #117: Problem: missing api of get_backup_mnemonics
add get_backup_mnemonics

### PR DESCRIPTION
**Pull request by leejw51crypto**
restore blueprint connections
update include
add get_backup_mnemonics, generate_mnemonics
added thirdparty library through lfs
use play-cpp-sdk v0.0.4-alpha

### CODE REVIEW

File: Source/ThirdParty/cronosplay/lib/Win64/.gitattributes [hunk 1]
```diff
@@ -0,0 +1,4 @@
+play_cpp_sdk.dll filter=lfs diff=lfs merge=lfs -text
+cxxbridge1.lib filter=lfs diff=lfs merge=lfs -text
+play_cpp_sdk.dll.lib filter=lfs diff=lfs merge=lfs -text
+play_cpp_sdk.lib filter=lfs diff=lfs merge=lfs -text
```
Comment (tomtau) [inline Source/ThirdParty/cronosplay/lib/Win64/.gitattributes:4]: if this are put directly in the repo, there should be a CI check to verify their checksum matches the release

File: Source/CronosPlayUnreal/Private/DefiWalletCoreActor.cpp [hunk 2]
```diff
@@ -406,6 +406,64 @@ void ADefiWalletCoreActor::InitializeNewDevelopmentOnlyWallet(
...
+    output = UTF8_TO_TCHAR(result.c_str());
+    success = true;
+  } catch (const rust::cxxbridge1::Error &e) {
+    success = false;
+    output = TEXT("");
+    output_message = FString::Printf(
+        TEXT("CronosPlayUnreal DevelopmentOnlyGenerateMnemonics Error: %s"),
+        UTF8_TO_TCHAR(e.what()));
+  }
+}
```
Comment (tomtau) [inline Source/CronosPlayUnreal/Private/DefiWalletCoreActor.cpp:465]: possibly some notes to be careful about the usage + help to zeroize those strings that aren't copied back to the caller

**Review by tomtau — CHANGES_REQUESTED** [review]
I haven't verified if those Git LFS files match the expected release ones -- better to add an automated check

File: Source/CronosPlayUnreal/Private/DefiWalletCoreActor.cpp (hunk 2 above)
Comment (leejw51crypto) [reply Source/CronosPlayUnreal/Private/DefiWalletCoreActor.cpp:465]: ok

File: Source/ThirdParty/cronosplay/lib/Win64/.gitattributes (hunk 1 above)
Comment (leejw51crypto) [reply Source/ThirdParty/cronosplay/lib/Win64/.gitattributes:4]: i'll check to ensure checksum

Comment (damoncro) [reply Source/ThirdParty/cronosplay/lib/Win64/.gitattributes:4]: Including the libraries makes cloning slow and repo big, and difficult to verify them valid or not (if someone injects malicious codes in it). Seems no big benefits... Please reconsider that.

Comment (damoncro) [reply Source/ThirdParty/cronosplay/lib/Win64/.gitattributes:4]: Besides their maybe many in the future, android, linux, ios, etc. It is difficult to include them all.

**Review by damoncro — APPROVED** [review]

File: Source/ThirdParty/cronosplay/lib/Win64/.gitattributes (hunk 1 above)
Comment (leejw51crypto) [reply Source/ThirdParty/cronosplay/lib/Win64/.gitattributes]: ok,removed binaries
we can release as current paradigm.

File: Source/CronosPlayUnreal/Private/DefiWalletCoreActor.cpp (hunk 2 above)
Comment (leejw51crypto) [reply Source/CronosPlayUnreal/Private/DefiWalletCoreActor.cpp:465]: working on now

Comment (leejw51crypto) [reply Source/CronosPlayUnreal/Private/DefiWalletCoreActor.cpp]: added warning, zeroize

File: Source/CronosPlayUnreal/Private/DefiWalletCoreActor.cpp [hunk 3]
```diff
@@ -396,6 +405,9 @@ void ADefiWalletCoreActor::InitializeNewDevelopmentOnlyWallet(
     rust::cxxbridge1::String result =
         _coreWallet->get_address(CoinType::CryptoOrgMainnet, 0);
     output = UTF8_TO_TCHAR(result.c_str());
+    // zeroize mnemonics
+    std::memset((void *)result.data(), 0, result.size());
```
Comment (tomtau) [inline Source/CronosPlayUnreal/Private/DefiWalletCoreActor.cpp:409]: `memset` may be removed by the compiler: https://en.cppreference.com/w/c/string/byte/memset

`memset_s` should be more reliable or `SecureZeroMemory`: https://docs.microsoft.com/en-us/previous-versions/windows/desktop/legacy/aa366877(v=vs.85)?redirectedfrom=MSDN

File: Source/CronosPlayUnreal/Private/DefiWalletCoreActor.cpp [hunk 4]
```diff
@@ -406,6 +418,75 @@ void ADefiWalletCoreActor::InitializeNewDevelopmentOnlyWallet(
...
+      break;
+    default:
+      throw "Invalid Word Count";
+      break;
+    }
+    rust::String result =
+        generate_mnemonics(TCHAR_TO_UTF8(*password), mywordcount);
+
+    output = UTF8_TO_TCHAR(result.c_str());
+    success = true;
```
Comment (tomtau) [inline Source/CronosPlayUnreal/Private/DefiWalletCoreActor.cpp:480]: zeroize result?

**Review by tomtau — APPROVED** [review]
`memset` may be optimized away by the compiler

File: Source/CronosPlayUnreal/Private/DefiWalletCoreActor.cpp (hunk 3 above)
Comment (tomtau) [reply Source/CronosPlayUnreal/Private/DefiWalletCoreActor.cpp:409]: > memset may be optimized away (under the [as-if](https://en.cppreference.com/w/c/language/as_if) rules) if the object modified by this function is not accessed again for the rest of its lifetime (e.g., [gcc bug 8537](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=8537)). For that reason, this function cannot be used to scrub memory (e.g., to fill an array that stored a password with zeroes). This optimization is prohibited for memset_s: it is guaranteed to perform the memory write.

File: Source/CronosPlayUnreal/Private/DefiWalletCoreActor.cpp (hunk 4 above)
Comment (leejw51crypto) [reply Source/CronosPlayUnreal/Private/DefiWalletCoreActor.cpp:480]: ok, i'll fix

File: Source/CronosPlayUnreal/Private/DefiWalletCoreActor.cpp (hunk 3 above)
Comment (leejw51crypto) [reply Source/CronosPlayUnreal/Private/DefiWalletCoreActor.cpp:409]: memset_s seems better, i'll check

**Pull request merged**